python/flow_sdk/shim_singer_sdk.py

```diff
@@ -92,14 +168,36 @@ def discover(self, discover: request.Discover) -> response.Discovered:
                     for p in entry.key_properties
                 ]
             else:
+                logger.info(f"Stream {stream.name} does not have a primary key, using '/_meta/row_id' as the key.") 
+                if entry.replication_method == REPLICATION_INCREMENTAL:
+                    logger.warn(f"Stream {stream.name} is using {REPLICATION_INCREMENTAL} replication without a primary key. If this is not an append-only data source, updates to existing documents will be captured as inserts with new keys. Switch to {REPLICATION_FULL_TABLE} to fix this.")
```

Comment:
🤔 I think it's very likely we'll want to tone down this warning.

What's a connector supposed to do to not be scary if it _wants_ this behavior?

ETA: Oh, I suppose it could use `/_meta/row_id` as the key explicitly? But for connector ergonomics, I'm not sure we should be asking that of users / connector developers.

Reply:
I guess it seemed like a decent sized footgun to me in that if you accidentally'd the key it would appear to work but generate incorrect data at runtime, and only after a connector restart at that, making it hard to track down. 

But yeah, not having a way to silence the warning is annoying. If you think it's common enough, we can just document it somewhere and get rid of the warning.

Reply:
Removed